scripts/tests/test_zkai_native_d128_two_slice_outer_backend_gate.py

```
@@ -0,0 +1,240 @@
+            with self.assertRaisesRegex(GATE.NativeD128TwoSliceOuterBackendError, "repo-relative"):
+                GATE.write_outputs(payload, pathlib.Path(raw_tmp) / "bad.json", None)
+        with self.assertRaisesRegex(GATE.NativeD128TwoSliceOuterBackendError, "without traversal"):
+            GATE.write_outputs(payload, pathlib.Path("docs/engineering/../bad.json"), None)
+
+    def test_write_outputs_cleans_temp_file_when_replace_fails(self) -> None:
+        payload = self.fresh_payload()
+        with tempfile.TemporaryDirectory(dir=ROOT) as raw_tmp:
+            tmp = pathlib.Path(raw_tmp)
+            json_path = (tmp / "native-two-slice.json").relative_to(ROOT)
+            with mock.patch.object(pathlib.Path, "replace", side_effect=OSError("forced replace failure")):
+                with self.assertRaisesRegex(OSError, "forced replace failure"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)